ibm/mas_devops/roles/cp4d_db2wh_restore/tasks/main.yml

```diff
@@ -0,0 +1,320 @@
+  shell: "oc cp /tmp/preRestore.sh {{ db2wh_pod_namespace }}/{{ db2wh_pod_name }}:/tmp/preRestore.sh"
+
+- name: Run preRestore.sh script on db2 pod {{ db2wh_pod_name }}
+  shell: oc exec -n {{ db2wh_pod_namespace }} {{ db2wh_pod_name }} -- su -lc '/tmp/preRestore.sh | tee /tmp/preRestore.log' db2inst1
+  register: prep_restore_output
+  failed_when: prep_restore_output.rc != 0
+
+- debug:
+    msg: "{{ prep_restore_output.stdout_lines }}"
+
+# 10. Run DB2 restore command
+# -----------------------------------------------------------------------------
+- name: Create restore.sh script in local /tmp
+  ansible.builtin.template:
+    src: restore.sh.j2
+    dest: /tmp/restore.sh
+    mode: '777'
+
+- name: Copy the restore.sh script into the db2 pod {{ db2wh_pod_name }}
+  shell: "oc cp /tmp/restore.sh {{ db2wh_pod_namespace }}/{{ db2wh_pod_name }}:/tmp/restore.sh"
+
+- name: "Run restore.sh script (this can take several minutes...)"
+  shell: oc exec -n {{ db2wh_pod_namespace }} {{ db2wh_pod_name }} -- su -lc '/tmp/restore.sh | tee /tmp/restore.log' db2inst1
+  register: restore_output
+  failed_when: restore_output.rc != 0 or 'SQL2581N' in restore_output.stdout_lines[0] # this db2 error code means something went wrong in restore command
+
+- debug:
+    msg: "{{ restore_output.stdout_lines }}"
+
+# 11. Run DB2 rollforward command
+# -----------------------------------------------------------------------------
+- name: Create rollforward.sh script in local /tmp
+  ansible.builtin.template:
+    src: rollforward.sh.j2
+    dest: /tmp/rollforward.sh
+    mode: '777'
+
+- name: Copy the rollforward.sh script into the db2 pod {{ db2wh_pod_name }}
+  shell: "oc cp /tmp/rollforward.sh {{ db2wh_pod_namespace }}/{{ db2wh_pod_name }}:/tmp/rollforward.sh"
+
+- name: "Run rollforward.sh script (this can take several minutes...)"
+  shell: oc exec -n {{ db2wh_pod_namespace }} {{ db2wh_pod_name }} -- su -lc '/tmp/rollforward.sh | tee /tmp/rollforward.log' db2inst1
+  register: rollforward_output
+  failed_when: rollforward_output.rc != 0 or 'SQL1119N' in rollforward_output.stdout_lines[0] # this db2 error code means something went wrong in rollforward command
+- debug:
+    msg: "{{ rollforward_output.stdout_lines }}"
+
+# 12. Activate DB2 after successfull rollforward
+# -----------------------------------------------------------------------------
+- name: Create postRestore.sh script in local /tmp
+  ansible.builtin.template:
+    src: postRestore.sh.j2
+    dest: /tmp/postRestore.sh
+    mode: '777'
+
+- name: Copy the postRestore.sh script into the db2 pod {{ db2wh_pod_name }}
+  shell: "oc cp /tmp/postRestore.sh {{ db2wh_pod_namespace }}/{{ db2wh_pod_name }}:/tmp/postRestore.sh"
+
+- name: Run postRestore.sh script on db2 pod {{ db2wh_pod_name }}
+  shell: oc exec -n {{ cpd_meta_namespace_target }} {{ db2wh_pod_name }} -- su -lc '/tmp/postRestore.sh | tee /tmp/postRestore.log' db2inst1
```

Comment:
cpd_meta_namespace_target must be replaced by db2wh_pod_namespace

Reply:
@ctrafa this is fixed